vignettes/workflows.Rmd

```diff
@@ -0,0 +1,292 @@
+
+By default, `plot_quality()` will plot the entire dataset:
+
+```{r}
+plot_quality(data)
+
+```
+
+Quality plots can also be run for a specific source:
+
+```{r}
+plot_quality(data, source = "/dutch2/DVA8K")
+```
+
+A quality plot consists of three separate visualizations, all designed to allow rapid visual inspection and spotting oddities:
+
+1. A density plot of turn durations. This is normally expected to look like a distribution that has a peak around 2000ms (2 seconds) and maximum lengths that do not far exceed 10000ms (10 seconds) (Liesenfeld & Dingemanse 2022). The goal of this plot is to allow eyeballing of oddities like turns of extreme durations or sets of turns with the exact same duration (unlikely in carefully segmented conversational data).
+
+2. A density plot of turn transition times. A plot like this is expected to look like a normal distribution centered around 0-200ms (Stivers et al. 2009). Deviations from this may signal problems in the dataset, for instance due to imprecise or automated annotation methods.
+
+3. A scatterplot of turn transition (x) by turn duration (y). This combines both distributions and is expected to look like a cloud of datapoints that is thickest in the middle region. Any standout patterns (for instance, turns whose duration is equal to their transition time) are indicative of problems in the segmentation or timing data.
+
+Each of the three plots can also be generated separately:
+```r
+
+plot_density(data, colname="duration", title="Turn durations",xlab="duration (ms)")
+
+plot_density(data, colname="FTO", title="Turn transitions (FTO)",xlab="FTO (ms)")
+
+plot_scatter(data, colname_x="FTO",colname_y="duration",title="Turn transitions and durations",xlab="transition (ms)", ylab="duration (ms)")
+
+```
+
+
+## Workflow 2: Plot conversations
+
+Another key use of `talkr` is to visualize conversational patterns. A first way to do so is `geom_turn()`, a ggplot2-compatible geom that visualizes the timing and duration of turns in a conversation.
+
+We can start by simply visualizing some of the conversations in the dataset. Here we sample the first four and plot the first minute of each. We display them together using `facet_wrap()` by `source`.
+
+```{r geom_turn_demon1}
+library(ggplot2)
+
+# we simplify participant names
+conv <- data |>
+  dplyr::group_by(source) |>
+  dplyr::mutate(participant = as.character(factor(participant, labels=c("A","B"),ordered=T)))
+
+# select first four conversations
+these_sources <- unique(data$source)[1:4]
+
+conv |>
+  dplyr::filter(end < 60000, # select first 60 seconds
+                source %in% these_sources) |> # filter to keep only these conversations
+  ggplot(aes(x = end, y = participant)) +
+  geom_turn(aes(
+    begin = begin,
+    end = end)) +
+  xlab("Time (ms)") +
+  ylab("") +
+  theme_turnPlot() +
+  facet_wrap(~source) # let's facet to show the conversations side by side
```

Comment:
This is a lot of extra code to demo a geom, and it's easy to get lost and miss the main point... Would it be an idea to generate a very dirty example first, and then clean it up with extra code?

Reply:
I would prefer to have #79 addressed before simplifying this; right now the messy plot is too messy